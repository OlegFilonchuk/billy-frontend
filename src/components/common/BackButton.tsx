import React, { FC } from 'react';
import { IoArrowBackOutline } from 'react-icons/all';
import { useHistory } from 'react-router-dom';
import Button from './Button';

const BackButton: FC = () => {
  const history = useHistory();

  const handleBack = () => {
    history.goBack();
  };

  return (
    <Button onClick={handleBack}>
      <IoArrowBackOutline />
    </Button>
  );
};

export default BackButton;
