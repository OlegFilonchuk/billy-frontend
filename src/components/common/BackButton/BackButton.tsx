import React, { FC } from 'react';
import { useTranslation } from 'react-i18next';
import { IoArrowBackOutline } from 'react-icons/io5';
import { useHistory } from 'react-router-dom';
import Button from '../Button';

const BackButton: FC = () => {
  const history = useHistory();

  const handleBack = () => {
    history.goBack();
  };

  const { t } = useTranslation();

  return (
    <Button onClick={handleBack} title={t('Go back')}>
      <IoArrowBackOutline />
    </Button>
  );
};

export default BackButton;
