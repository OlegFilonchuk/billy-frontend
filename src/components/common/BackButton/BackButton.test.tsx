import { fireEvent, RenderResult } from '@testing-library/react';
import React from 'react';
import renderWithI18n from '../../../utils/testUtills';
import BackButton from './BackButton';

const mock = jest.fn();

jest.mock('react-router-dom', () => ({
  useHistory: () => ({
    goBack: mock,
  }),
}));

describe('BackButton', () => {
  let ui: RenderResult;
  beforeEach(() => {
    ui = renderWithI18n(<BackButton />);
  });

  it('should render properly', () => {
    expect(ui.container).toMatchSnapshot();
  });

  it("should trigger history's goBack function", () => {
    const button = ui.getByTitle('Go back');

    fireEvent.click(button);
    expect(mock).toHaveBeenCalled();
  });
});
