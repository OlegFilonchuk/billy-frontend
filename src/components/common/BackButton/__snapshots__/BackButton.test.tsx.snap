// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackButton should render properly 1`] = `
<div>
  <button
    class="transition duration-150 hover:shadow-lg focus:shadow-lg focus:ring ring-accent disabled:opacity-50 p-2 rounded inline-flex justify-center items-center    "
    title="Go back"
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M244 400L100 256l144-144M120 256h292"
        fill="none"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="48"
      />
    </svg>
  </button>
</div>
`;
